test/tests/fips_test_suite/common.rs

```diff
@@ -237,11 +237,101 @@ pub fn mbx_send_and_check_resp_hdr<T: HwModel, U: FromBytes + AsBytes>(
     //Ok(U::read_from(resp_bytes.as_bytes()).unwrap())
 }
 
-// Returns true if not all bytes are the same
+fn get_cmd_id(dpe_cmd: &mut Command) -> u32 {
+    match dpe_cmd {
+        Command::GetProfile => Command::GET_PROFILE,
+        Command::InitCtx(_) => Command::INITIALIZE_CONTEXT,
+        Command::DeriveContext(_) => Command::DERIVE_CONTEXT,
+        Command::CertifyKey(_) => Command::CERTIFY_KEY,
+        Command::Sign(_) => Command::SIGN,
+        Command::RotateCtx(_) => Command::ROTATE_CONTEXT_HANDLE,
+        Command::DestroyCtx(_) => Command::DESTROY_CONTEXT,
+        Command::GetCertificateChain(_) => Command::GET_CERTIFICATE_CHAIN,
+    }
+}
+pub fn as_bytes(dpe_cmd: &mut Command) -> &[u8] {
+    match dpe_cmd {
+        Command::CertifyKey(cmd) => cmd.as_bytes(),
+        Command::DeriveContext(cmd) => cmd.as_bytes(),
+        Command::GetCertificateChain(cmd) => cmd.as_bytes(),
+        Command::DestroyCtx(cmd) => cmd.as_bytes(),
+        Command::GetProfile => &[],
+        Command::InitCtx(cmd) => cmd.as_bytes(),
+        Command::RotateCtx(cmd) => cmd.as_bytes(),
+        Command::Sign(cmd) => cmd.as_bytes(),
+    }
+}
+
+pub fn parse_dpe_response(dpe_cmd: &mut Command, resp_bytes: &[u8]) -> Response {
+    match dpe_cmd {
+        Command::CertifyKey(_) => {
+            Response::CertifyKey(CertifyKeyResp::read_from(resp_bytes).unwrap())
+        }
+        Command::DeriveContext(_) => {
+            Response::DeriveContext(DeriveContextResp::read_from(resp_bytes).unwrap())
+        }
+        Command::GetCertificateChain(_) => {
+            Response::GetCertificateChain(GetCertificateChainResp::read_from(resp_bytes).unwrap())
+        }
+        Command::DestroyCtx(_) => Response::DestroyCtx(ResponseHdr::read_from(resp_bytes).unwrap()),
+        Command::GetProfile => Response::GetProfile(GetProfileResp::read_from(resp_bytes).unwrap()),
+        Command::InitCtx(_) => Response::InitCtx(NewHandleResp::read_from(resp_bytes).unwrap()),
+        Command::RotateCtx(_) => Response::RotateCtx(NewHandleResp::read_from(resp_bytes).unwrap()),
+        Command::Sign(_) => Response::Sign(SignResp::read_from(resp_bytes).unwrap()),
+    }
+}
+
+pub fn execute_dpe_cmd<T: HwModel>(hw: &mut T, dpe_cmd: &mut Command) -> Response {
```

Comment:
nit: You could probably use https://github.com/chipsalliance/caliptra-sw/blob/cc31e5f5b349ba47c052648f3c529225f97962c0/runtime/tests/runtime_integration_tests/common.rs#L163

Although you might need to refactor and to share the implementation, so I'm fine if you want to defer this one.